RandBLAS/dense.hh

```diff
@@ -204,88 +204,97 @@ DenseSkOp<T,RNG>::~DenseSkOp() {
  *
  * @returns the updated CBRNG state
  */
-template <typename T, typename RNG, typename OP>
-RNGState<RNG> fill_rmat(
-    int64_t n_rows,
+   
+
+template<typename T, typename RNG, typename OP>
+auto fill_rsubmat_omp(
     int64_t n_cols,
-    T* mat,
-    const RNGState<RNG> & seed,
-    MajorAxis ma = MajorAxis::Long
+    T* smat,
+    int64_t n_srows,
+    int64_t n_scols,
+    int64_t ptr,
+    const RNGState<RNG> & seed
```

Comment:
I have pushed the new changes related to the documentation. Please let me know what you think.

In the current implementation for fill_rsubmat_omp, the only information related to the original, larger random matrix is the number of columns. The implementation doesn't check if the sub matrix goes out of bounds of the original random matrix. Would you like fill_rsubmat_omp to have an additional argument for the number of rows of the original random matrix so that at the beginning we can do a check to make sure the submatrix is within bounds?